python/pyspark/sql/session.py

```diff
@@ -952,12 +953,29 @@ def createDataFrame(  # type: ignore[misc]
             schema = [x.encode("utf-8") if not isinstance(x, str) else x for x in schema]
 
         try:
-            import pandas
+            import pandas as pd
 
             has_pandas = True
         except Exception:
             has_pandas = False
-        if has_pandas and isinstance(data, pandas.DataFrame):
+
+        try:
+            import numpy as np
+
+            has_numpy = True
+        except Exception:
+            has_numpy = False
+
+        if has_numpy and isinstance(data, np.ndarray):
+            from pyspark.sql.pandas.utils import require_minimum_pandas_version
+
+            require_minimum_pandas_version()
+            if data.ndim not in [1, 2]:
+                raise ImportError("NumPy array input should be of 1 or 2 dimensions.")
+            column_names = ["value"] if data.ndim == 1 else ["_1", "_2"]
```

Comment:
Would have to be done separately but just noting here: maybe we should make the case like `spark.createDataFrame([1, 2])` working too. `spark.createDataFrame([1, 2], "int")` works. the problem seems like in schema inference.

Reply:
We seem to infer the schema of the input list from the first element only. To support `spark.createDataFrame([1, 2])`, we may assume the elements in the list are of the same type. Is that acceptable?

Reply:
Also, the input list may consist of both NumPy scalars and native Python scalars, e.g. `spark.createDataFrame([np.int8(1), 2])`.

Reply:
Ideally it infers the narrowest type supporting all values. I think there is another issue open about this

Reply:
I see, then we may have to iterate through the values and find the Lowest Common Ancestor of all involved types.